pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -107,16 +108,30 @@ func nodeConditionFinder(status *corev1.NodeStatus, condType corev1.NodeConditio
 	return nil
 }
 
+func isNodeSchedulable(node *corev1.Node) bool {
+	for _, taint := range node.Spec.Taints {
+		if taint.Effect == corev1.TaintEffectNoSchedule && taint.Key == corev1.TaintNodeUnschedulable {
+			return false
+		}
+	}
+	return true
+}
+
 func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
 	klog.V(5).Info("Syncing guards")
 
 	if c.createConditionalFunc == nil {
-		return fmt.Errorf("create conditional not set")
+		return fmt.Errorf("createConditionalFunc not set")
 	}
 
-	shouldCreate, err := c.createConditionalFunc()
+	shouldCreate, unknown, err := c.createConditionalFunc()
```

Comment:
`notReady`? `notSynced` ?

Reply:
The `createConditionalFunc` method is a black box for the guard controller. Being ready or synced is too specific without a context.  What about `conditionIncomplete`?